spec/compiler/macro/macro_methods_spec.cr

```diff
@@ -582,6 +582,10 @@ module Crystal
         assert_macro "", %({{[1, 2, 3].reduce(4) { |acc, val| acc * val }}}), [] of ASTNode, "24"
       end
 
+      it "executes reduce with initial value of other type" do
```

Comment:
I'd just merge this example with the one above. It can simply have to expectations.